### ISSUE TITLE
Environment variables not picked up by signing plugin

### ISSUE THREAD

**Issue by jeffreydelooff** [vanniktech/gradle-maven-publish-plugin#149]
In the root `build.gradle.kts` I have defined a signing block (as per the gradle documentation) to retrieve environment variables (starting with `ORG_GRADLE_PROJECT_`) for the signing process:

```
signing {
    val signingKeyId: String? by project
    val signingKey: String? by project
    val signingPassword: String? by project
    useInMemoryPgpKeys(signingKeyId, signingKey, signingPassword)
    sign(project.publishing.publications)
}
```

However, the Sign instance cannot find any signatories:

```
Caused by: org.gradle.api.InvalidUserDataException: Cannot perform signing task ':interactor:signMavenPublication' because it has no configured signatory
        at org.gradle.plugins.signing.Sign.generate(Sign.java:202)
        at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:104)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.doExecute(StandardTaskAction.java:49)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:42)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:28)
```

Is there a way for this plugin to read the environment variables at all? Or is it required to provide them as gradle properties directly?

Currently the environment variables are provided in a CI setting with Github secrets. The project is open source: https://github.com/elementsinteractive/android-mobilization/runs/882074558?check_suite_focus=true#step:11:272

**Comment by vanniktech** [comment]
We're just delegating to Gradle, is that an issue there?

**Comment by gabrielittner** [comment]
Do you have a link to where you defined the `signing { ... }` block in your repository? I can't see it anywhere.

**Comment by gabrielittner** [comment]
Closing for now. If you have a branch where it's failing I can take another look

**Comment by jeffreydelooff** [comment]
I haven't been able to get this to work yet. I have updated it to match the set-up in this repository but still gives the same exception.

I have added the signing block here https://github.com/elementsinteractive/android-mobilization/blob/main/build.gradle.kts#L94

and added a release work flow here: https://github.com/elementsinteractive/android-mobilization/blob/main/.github/workflows/publish-release.yml

**Comment by gabrielittner** [comment]
Can you try `project.hasProperty` and `project.getProperty`? It seems like within the `signing` block you might need to explicitly write it like that. The in memory signing key setup here wasn't actually tested so I need to fix it here as well.

**Comment by gabrielittner** [comment]
@jeffreydelooff I fixed the signing setup in this repo in #174

**Comment by jeffreydelooff** [comment]
Thank you for the info. I have updated the project but didn't help.

I just noticed the plugin outputs `No compatible plugin found in project..` for a pure Kotlin library.
Would I also need to apply `java-library`?

**Comment by gabrielittner** [comment]
Sorry, for the late reply. Yes currently you also need `java-library`. We use that as an identifier for the default jvm publishing process. It could be possible that we can get rid of that requirement or switch it to the java base plugin which is applied automatically by the kotlin plugin

**Comment by jeffreydelooff** [comment]
No problem. I got it working on Friday. This plugin was also applying to the root project so I updated it to this:
`id("com.vanniktech.maven.publish") version Versions.mavenPublish apply false`. I have also added a [SigningMetaModulePlugin](https://github.com/elementsinteractive/android-mobilization/blob/main/buildSrc/src/main/java/nl/elements/mobilization/SigningMetaModulePlugin.kt) to help with the signing set-up only when your plugin is applied. I believe my secrets in Github were also wrong (removed all newline chars \n and copied the private key as is by only using: `gpg --armor --export-secret-keys <ID>`
